### PR TITLE
feat(spark-core): add core support for jwt auth

### PR DESCRIPTION
**Pull request by ianwremmel**

### CODE REVIEW

File: packages/spark-core/src/plugins/credentials/credentials/common.js [hunk 1]
```diff
@@ -218,6 +227,29 @@ export default {
   },
 
   @oneFlight
+  requestAccessTokenFromJwt(options) {
+    return this.spark.request({
+      method: `POST`,
+      // I'm not thrilled by directly referencin the hydra service url, but
```
Comment (mattnorris) [inline packages/spark-core/src/plugins/credentials/credentials/common.js]: "referencing"

File: srcdist.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
```
Comment (mattnorris) [inline srcdist.sh]: Isn't this the same as `distsrc.sh` above?

Comment (ianwremmel) [inline srcdist.sh]: Nope, it should be exactly the opposite.

File: packages/spark-core/src/plugins/credentials/credentials/common.js [hunk 3]
```diff
@@ -35,6 +35,9 @@ export default {
     canRefresh: {
       deps: [`authorization.canRefresh`],
       fn() {
+        if (this.config.requestJWT) {
+          return true;
+        }
```
Comment (saurjai3) [inline packages/spark-core/src/plugins/credentials/credentials/common.js]: can be change to return this.config.requestJWT

**Comment by saurjai3** [comment]
LGTM

**Pull request merged**